### PR TITLE
HPCC-17900 Ability to capture git status through eclcc

### PR DESCRIPTION
**Pull request by richardkchapman**
Add checkDirty flag to ecl command - since it's been documented that way.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] There is an existing JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

## Testing:
Observed help text change...

### CODE REVIEW

**Comment by richardkchapman** [comment]
@afishbeck Please review

File: ecl/eclcmd/eclcmd_common.hpp [hunk 1]
```diff
@@ -167,6 +167,7 @@ typedef IEclCommand *(*EclCommandFactory)(const char *cmdname);
 #define ECLOPT_MANIFEST_DASH "-manifest"
 #define ECLOPT_LEGACY "--legacy"
 #define ECLOPT_LEGACY_DASH "-legacy"
+#define ECLOPT_CHECKDIRTY "-checkDirty"
```
Comment (afishbeck) [inline ecl/eclcmd/eclcmd_common.hpp]: Two dashes is generally preferred for long form eclcmd option names.  Legacy and manifest are weird cases where for some reason I support both forms.  Could support both for this too, but I think that was originally thinking people would be used to the eclcc version, but it usually isn't the case that eclcmd users also use eclcc directly.

File: ecl/eclcmd/eclcmd_common.hpp [hunk 2]
```diff
@@ -261,7 +262,7 @@ class EclCmdCommon : implements IEclCommand, public CInterface
 class EclCmdWithEclTarget : public EclCmdCommon
 {
 public:
-    EclCmdWithEclTarget() : optLegacy(false), optNoArchive(false), optResultLimit((unsigned)-1), optDebug(false), paramCount(0)
+    EclCmdWithEclTarget() : optLegacy(false), optCheckDirty(false), optNoArchive(false), optResultLimit((unsigned)-1), optDebug(false), paramCount(0)
```
Comment (afishbeck) [inline ecl/eclcmd/eclcmd_common.hpp]: Should add --checkDirty to EclCmdWithEclTarget::usage().

**Comment by afishbeck** [comment]
@richardkchapman ignore my initial comments.  I forgot about the common usage generation for eclcc options in eclcmd.  Looks good.

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge

**Pull request merged**